apps/api/tests/test_small_modules_coverage.py

```
@@ -0,0 +1,616 @@
+        logger.handlers = prev_handlers
+        logger.propagate = prev_propagate
+        logger.setLevel(prev_level)
+        if prev_configured:
+            setattr(logger, "_reframe_configured", True)
+        elif hasattr(logger, "_reframe_configured"):
+            delattr(logger, "_reframe_configured")
+
+
+def test_setup_logging_returns_early_when_already_configured():
+    logger = logging.getLogger("reframe")
+    setattr(logger, "_reframe_configured", True)
```

Ruff (0.15.15)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)